src/main/java/org/wildfly/security/util/ProviderServiceLoaderSupplier.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.util;
+
+import static org.wildfly.security.util.ProviderUtil.INSTALLED_PROVIDERS;
+
+import org.wildfly.security.WildFlyElytronProvider;
+import org.wildfly.security.util._private.Arrays2;
+
+import java.security.Provider;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
+
+/**
+ * A supplier which uses a service loader to find all {@link Provider} instances that aren't in the list of
+ * installed security providers and returns them as an array. The result is then cached.
+ *
+ * @author <a href="mailto:fjuma@redhat.com">Farah Juma</a>
+ * @since 1.2.2
+ */
+public class ProviderServiceLoaderSupplier extends ServiceLoaderSupplier<Provider> {
+
+    private static final HashSet INSTALLED_PROVIDERS_SET;
```

Comment:
This raw type should have a type of ``HashSet<Class<?>>`` or ``Set<Class<?>>``.

Also it should not be static.  The set of installed providers may change over time, so the set of types of providers can also change over time.  Therefore the set should be reconstructed each time the ``INSTALLED_PROVIDERS`` supplier is evaluated via its ``get()`` method.

Reply:
Good catch, will fix this.